code/clue.js

```diff
@@ -1,59 +1,251 @@
+  "library",
+  "billiardroom",
+  "lounge",
+  "ballroom",
+  "hall",
+  "spa",
+  "livingroom",
+  "observatory",
+  "theater",
+  "guesthouse",
+  "patio",
+];
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+// function that randomly selecting one item from the array that you pass into the function
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+//object mystery that will store new random values
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null,
+  //favouriteWeapon: null,
+};
 
-// This function will be invoked when you click on the killer card.
+//shuffleWeapon function
+const shuffleWeapon = (arrayOfSuspects) => {
+  arrayOfSuspects.forEach((suspect) => {
+    suspect.favouriteWeapon = randomSelector(weapons).id;
+    //console.log(suspect.favouriteWeapon);
+  });
+};
+shuffleWeapon(suspects);
+
+// This function randomly selects a killer from the suspects. And add that to the mystery object.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  //mystery.favouriteWeapon = shuffleWeapon(suspects);
+  mystery.killer = randomSelector(suspects);
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer,age,occupation and description
+  document.getElementById("killerCard").style.background = mystery.killer.color;
+  document.getElementById("killerImage").src = mystery.killer.image;
+  document.getElementById(
+    "killerName"
+  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById("killerAge").innerHTML = `${mystery.killer.age}`;
+  document.getElementById(
+    "killerOccupation"
+  ).innerHTML = `${mystery.killer.occupation}`;
+  document.getElementById(
+    "killerDescription"
+  ).innerHTML = `${mystery.killer.description}`;
+  document.getElementById(
+    "killerFavouriteWeapon"
+  ).innerHTML = `Loves to use their ${mystery.killer.favouriteWeapon}`;
+};
+
+//when you click on the killerCard card it will call pickKiller function
+
+document.getElementById("killerCard").onclick = pickKiller;
+
+// This will randomly select a weapon from the weapons. And add that to the mystery object.
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+
+  //This will show the weapon name and weight on the card
+
+  document.getElementById("weaponCard").style.background = mystery.weapon.color;
+
+  document.getElementById("weaponName").innerHTML = `${mystery.weapon.name}`;
```

Comment:
` document.getElementById("weaponName").innerHTML = mystery.weapon.name;`
Also works here since there is nothing in the string template